### PR TITLE
[AI CHAT] CT-178: Add copy paste button to Aichat level

### PR DESCRIPTION
**Pull request by ebeastlake**
This PR:
- Adds a CopyButton that allows a user to copy their chat history.
- Adds a timestamp field to the messages.
- Makes an error message the default rendered chat message if the response from the server is empty.

## Links

https://codedotorg.atlassian.net/browse/CT-178

## Testing story

Tested locally.

A screenshot of the page, including the new copy button in the upper right-hand corner and a confirmation when copying has succeeded.
Screenshot 2023-11-16 at 2 51 15 PM

The error message is now used as default instead of the "too personal" message. 
Screenshot 2023-11-16 at 10 41 17 PM

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/aichat/redux/aichatRedux.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
+import moment from 'moment';
 import {createSlice, PayloadAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {LabState} from '@cdo/apps/lab2/lab2Redux';
+const registerReducers = require('@cdo/apps/redux').registerReducers;
+
+import {initialChatMessages} from '../constants';
+import {getChatCompletionMessage} from '../chatApi';
```
Comment (ebeastlake) [inline apps/src/aichat/redux/aichatRedux.ts:7]: I reorganize imports so the absolute ones are at the top and the relative ones are below, sorry for the extra diff 🙈

File: apps/src/aichat/redux/aichatRedux.ts [hunk 2]
```diff
@@ -86,6 +89,9 @@ export const submitChatMessage = createAsyncThunk(
         role: Role.ASSISTANT,
         status: Status.OK,
         chatMessageText: chatApiResponse.assistantResponse,
+        // The accuracy of this timestamp is debatable since it's not when our backend
+        // issued the message, but it's good enough for user testing.
```
Comment (Erin007) [inline apps/src/aichat/redux/aichatRedux.ts:93]: 👍

File: apps/src/aichat/views/CopyButton.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    );
+  };
+
+  return (
+    <Button
+      color={Button.ButtonColor.white}
+      key="copy"
+      onClick={() => handleCopy()}
+      size={Button.ButtonSize.small}
+      text="Copy Conversation History"
```
Comment (Erin007) [inline apps/src/aichat/views/CopyButton.tsx:38]: we could maybe add  `icon="clipboard"` but this is intended to be so short-lived totally up to you!

**Review by Erin007 — APPROVED** [review]
Quick, clever solution! Thanks @ebeastlake 🚀

File: apps/src/aichat/views/CopyButton.tsx (hunk 3 above)
Comment (ebeastlake) [reply apps/src/aichat/views/CopyButton.tsx:38]: I had that initially when the button was in a different location, but Michael suggested removing it. Now that I see that both "Version History" and "Show Code" have icons, I'll re-add the icon for consistency!

**Pull request merged**